### PR TITLE
Addressing #46

### PR DESCRIPTION
**Pull request by zhiweigan**

### CODE REVIEW

File: fluid-music/test/test-client.js [hunk 1]
```diff
@@ -67,6 +93,24 @@ describe('fluid.Client (TCP/IPC Client)', function() {
     return result.should.be.rejected();
   }));
 
+  it('When server returns an invalid response, the promise returned by client.send should be rejected', (function() {
+    const client = new fluid.Client(22354);
+    const result = client.send(fluid.audiotrack.select('hi'));
+    return result.should.be.rejected();
+  }));
+
+  it('When the client sends an incomplete message, the promise should be rejected', (function() {
```
Comment (CharlesHolbrow) [inline fluid-music/test/test-client.js:102]: Should this be "when the server replies with an incomplete message"?

Comment (CharlesHolbrow) [reply fluid-music/test/test-client.js:102]: Both sides should be tested. (when the client sends an incomplete message, and when the server replies with an incomplete message)

**Comment by zhiweigan** [comment]
I've added the suggested tests, and fixed an issue where if the client connects before ```.send``` is called, the promise would have been rejected.

**Comment by zhiweigan** [comment]
Should be all good now!

**Comment by CharlesHolbrow** [comment]
This is looking pretty good, so I'm merging it in. In the long term there are a few things that will be worth fixing. I think these changes should wait until they are needed, or a larger overhaul is warranted. 

Tests shouldn't "reach in" to the objects they are testing (`if(client.client._socket.destroyed) done();`). They should only test the interface, not the implementation. If the implementation changes, we don't want it to break the tests:
https://github.com/CharlesHolbrow/cybr/blob/faea61c069360ce5221c7e4df3baa6ace29ef382/fluid-music/test/test-client.js#L97-L104

Right now the behavior of calling send twice on a client is undefined. It should probably throw when we call `send` a second time on an open client. However, eventually, we should support this behavior with proper `.listen` `.close` and promise queues etc. So it's probably not worth testing for this now, even though there are probably some sublte bugs (i.e. `cli.js`)

**Pull request merged**